2021-06-29 20:12:41.558383: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcuda.so.1
2021-06-29 20:12:41.585324: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1561] Found device 0 with properties: 
pciBusID: 0000:65:00.0 name: Quadro P5000 computeCapability: 6.1
coreClock: 1.7335GHz coreCount: 20 deviceMemorySize: 15.89GiB deviceMemoryBandwidth: 269.00GiB/s
2021-06-29 20:12:41.585591: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1
2021-06-29 20:12:41.587830: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10
2021-06-29 20:12:41.589709: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10
2021-06-29 20:12:41.590084: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10
2021-06-29 20:12:41.591913: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10
2021-06-29 20:12:41.592924: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10
2021-06-29 20:12:41.596406: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7
2021-06-29 20:12:41.597047: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1703] Adding visible gpu devices: 0
2021-06-29 20:12:41.597352: I tensorflow/core/platform/cpu_feature_guard.cc:143] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 AVX512F FMA
2021-06-29 20:12:41.603514: I tensorflow/core/platform/profile_utils/cpu_utils.cc:102] CPU Frequency: 3600000000 Hz
2021-06-29 20:12:41.603949: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x7f5fc8000b60 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2021-06-29 20:12:41.603974: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
2021-06-29 20:12:41.666825: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x55d252f18c60 initialized for platform CUDA (this does not guarantee that XLA will be used). Devices:
2021-06-29 20:12:41.666879: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Quadro P5000, Compute Capability 6.1
2021-06-29 20:12:41.667703: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1561] Found device 0 with properties: 
pciBusID: 0000:65:00.0 name: Quadro P5000 computeCapability: 6.1
coreClock: 1.7335GHz coreCount: 20 deviceMemorySize: 15.89GiB deviceMemoryBandwidth: 269.00GiB/s
2021-06-29 20:12:41.667796: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1
2021-06-29 20:12:41.667828: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10
2021-06-29 20:12:41.667857: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10
2021-06-29 20:12:41.667886: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10
2021-06-29 20:12:41.667914: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10
2021-06-29 20:12:41.667943: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10
2021-06-29 20:12:41.667971: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7
2021-06-29 20:12:41.668960: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1703] Adding visible gpu devices: 0
2021-06-29 20:12:41.669030: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1
2021-06-29 20:12:41.669759: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1102] Device interconnect StreamExecutor with strength 1 edge matrix:
2021-06-29 20:12:41.669792: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1108]      0 
2021-06-29 20:12:41.669807: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1121] 0:   N 
2021-06-29 20:12:41.670968: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1247] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 345 MB memory) -> physical GPU (device: 0, name: Quadro P5000, pci bus id: 0000:65:00.0, compute capability: 6.1)
2021-06-29 20:12:46.259971: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10
2021-06-29 20:12:46.396350: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7
2021-06-29 20:12:46.667254: E tensorflow/stream_executor/cuda/cuda_dnn.cc:329] Could not create cudnn handle: CUDNN_STATUS_INTERNAL_ERROR
2021-06-29 20:12:46.671814: E tensorflow/stream_executor/cuda/cuda_dnn.cc:329] Could not create cudnn handle: CUDNN_STATUS_INTERNAL_ERROR
NUM_ITER 70
BATCH_SIZE 50
NUM_BATCH 100
NUM_SNPS 36
L 50000
NUM_CLASSES 2
NUM_CHANNELS 2
NUM_ITER 300
BATCH_SIZE 50
NUM_BATCH 100
NUM_SNPS 36
L 50000
NUM_CLASSES 2
NUM_CHANNELS 2
{'model': 'exp', 'params': 'N1,N2,growth,T1,T2', 'data_h5': '/bigdata/smathieson/pg-gan/1000g/HDF5/CHB.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.h5', 'bed': '/bigdata/smathieson/pg-gan/1000g/HDF5/20120824_strict_mask.bed', 'reco_folder': None, 'grid': None, 'toy': None, 'seed': 1833}
N1,N2,growth,T1,T2
[<util.Parameter object at 0x7f6114287b50>, <util.Parameter object at 0x7f6188b36150>, <util.Parameter object at 0x7f61145f0d90>, <util.Parameter object at 0x7f61142a8b90>, <util.Parameter object at 0x7f61142a8ad0>]
['N1', 'N2', 'growth', 'T1', 'T2']
NAME	VALUE	MIN	MAX
N1	9000	1000	30000
NAME	VALUE	MIN	MAX
N2	5000	1000	30000
NAME	VALUE	MIN	MAX
growth	0.005	0.0	0.05
NAME	VALUE	MIN	MAX
T1	2000	1500	5000
NAME	VALUE	MIN	MAX
T2	350	100	1500
['calldata', 'variants']
['GT'] ['CHROM', 'POS']
raw (11703947, 99, 2)
after haps (11703947, 198)
parameters <class 'list'> [<util.Parameter object at 0x7f6114287b50>, <util.Parameter object at 0x7f6188b36150>, <util.Parameter object at 0x7f61145f0d90>, <util.Parameter object at 0x7f61142a8b90>, <util.Parameter object at 0x7f61142a8ad0>]
Model: "one_pop_model"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
conv2d (Conv2D)              (None, 198, 32, 32)       352       
_________________________________________________________________
conv2d_1 (Conv2D)            (None, 198, 12, 64)       10304     
_________________________________________________________________
max_pooling2d (MaxPooling2D) multiple                  0         
_________________________________________________________________
flatten (Flatten)            (None, 384)               0         
_________________________________________________________________
dropout (Dropout)            (None, 128)               0         
_________________________________________________________________
dense (Dense)                (None, 128)               49280     
_________________________________________________________________
dense_1 (Dense)              (None, 128)               16512     
_________________________________________________________________
dense_2 (Dense)              (None, 4)                 516       
_________________________________________________________________
dense_3 (Dense)              (None, 1)                 5         
=================================================================
Total params: 76,969
Trainable params: 76,969
Non-trainable params: 0
_________________________________________________________________
(198, 36, 2)
pretraining starts
at iteration  1
Traceback (most recent call last):
  File "pg_gan.py", line 423, in <module>
    main()
  File "pg_gan.py", line 65, in main
    iterator, parameters, opts.seed, toy=opts.toy)
  File "pg_gan.py", line 162, in simulated_annealing
    s_current = pg_gan.disc_pretraining(800, BATCH_SIZE)
  File "pg_gan.py", line 317, in disc_pretraining
    VAE_model.run_training()
  File "/homes/tdo/pg-gan2/VAE_train.py", line 45, in run_training
    self.training_sa(VAE_NUM_BATCH, VAE_BATCH_SIZE) 
  File "/homes/tdo/pg-gan2/VAE_train.py", line 37, in training_sa
    loss = train_step(self.model, real_regions, optimizer)
  File "/packages/cs/python3.7.7/lib/python3.7/site-packages/tensorflow/python/eager/def_function.py", line 580, in __call__
    result = self._call(*args, **kwds)
  File "/packages/cs/python3.7.7/lib/python3.7/site-packages/tensorflow/python/eager/def_function.py", line 644, in _call
    return self._stateless_fn(*args, **kwds)
  File "/packages/cs/python3.7.7/lib/python3.7/site-packages/tensorflow/python/eager/function.py", line 2420, in __call__
    return graph_function._filtered_call(args, kwargs)  # pylint: disable=protected-access
  File "/packages/cs/python3.7.7/lib/python3.7/site-packages/tensorflow/python/eager/function.py", line 1665, in _filtered_call
    self.captured_inputs)
  File "/packages/cs/python3.7.7/lib/python3.7/site-packages/tensorflow/python/eager/function.py", line 1746, in _call_flat
    ctx, args, cancellation_manager=cancellation_manager))
  File "/packages/cs/python3.7.7/lib/python3.7/site-packages/tensorflow/python/eager/function.py", line 598, in call
    ctx=ctx)
  File "/packages/cs/python3.7.7/lib/python3.7/site-packages/tensorflow/python/eager/execute.py", line 60, in quick_execute
    inputs, attrs, num_outputs)
tensorflow.python.framework.errors_impl.UnknownError:  Failed to get convolution algorithm. This is probably because cuDNN failed to initialize, so try looking to see if a warning log message was printed above.
	 [[node encoder/conv2d_2/Conv2D (defined at /homes/tdo/pg-gan2/VAE.py:24) ]] [Op:__inference_train_step_1632]

Errors may have originated from an input operation.
Input Source operations connected to node encoder/conv2d_2/Conv2D:
 truediv (defined at /homes/tdo/pg-gan2/VAE.py:111)

Function call stack:
train_step

